File: pes/client/ui/src/services/activityService.js
import apiClient from './apiClient';
import { API_ENDPOINTS, handleApiResponse, handleApiError } from './index';

export const activityService = {
  async getAllActivities() {
    try {
      const response = await apiClient.get(API_ENDPOINTS.ACTIVITIES.LIST);
      return handleApiResponse(response);
    } catch (error) {
      return handleApiError(error);
    }
  },

  async getActivityById(activityId) {
    try {
      const response = await apiClient.get(API_ENDPOINTS.ACTIVITIES.GET_BY_ID(activityId));
      return handleApiResponse(response);
    } catch (error) {
      return handleApiError(error);
    }
  },

  async createActivity(activityData) {
    try {
      const response = await apiClient.post(API_ENDPOINTS.ACTIVITIES.CREATE, activityData);
      return handleApiResponse(response);
    } catch (error) {
      return handleApiError(error);
    }
  },

  async updateActivity(activityId, activityData) {
    try {
      const response = await apiClient.put(API_ENDPOINTS.ACTIVITIES.UPDATE(activityId), activityData);
      return handleApiResponse(response);
    } catch (error) {
      return handleApiError(error);
    }
  },

  async deleteActivity(activityId) {
    try {
      const response = await apiClient.delete(API_ENDPOINTS.ACTIVITIES.DELETE(activityId));
      return handleApiResponse(response);
    } catch (error) {
      return handleApiError(error);
    }
  },

  async getActivitiesByScheduleId(scheduleId) {
    try {
      const response = await apiClient.get(API_ENDPOINTS.ACTIVITIES.BY_SCHEDULE(scheduleId));
      return handleApiResponse(response);
    } catch (error) {
      return handleApiError(error);
    }
  },

  async getActivitiesByClassId(classId) {
    try {
      const response = await apiClient.get(API_ENDPOINTS.ACTIVITIES.BY_CLASS(classId));
      return handleApiResponse(response);
    } catch (error) {
      return handleApiError(error);
    }
  },

  async getActivitiesByClassAndDay(classId, dayOfWeek) {
    try {
      const response = await apiClient.get(API_ENDPOINTS.ACTIVITIES.BY_CLASS_AND_DAY(classId, dayOfWeek));
      return handleApiResponse(response);
    } catch (error) {
      return handleApiError(error);
    }
  },

  async assignActivityToClass(assignmentData) {
    try {
      const response = await apiClient.post('/activities/assign', assignmentData);
      return handleApiResponse(response);
    } catch (error) {
      return handleApiError(error);
    }
  },
};

File: assignment_2/be/src/main/java/com/sba301/group1/pes_be/services/ActivityService.java
package com.sba301.group1.pes_be.services;

import com.sba301.group1.pes_be.requests.AssignActivityToClassRequest;
import com.sba301.group1.pes_be.requests.CreateActivityRequest;
import com.sba301.group1.pes_be.requests.ResponseObject;
import com.sba301.group1.pes_be.requests.UpdateActivityRequest;
import org.springframework.http.ResponseEntity;

public interface ActivityService {
    
    ResponseEntity<ResponseObject> createActivity(CreateActivityRequest request);
    
    ResponseEntity<ResponseObject> updateActivity(Integer activityId, UpdateActivityRequest request);
    
    ResponseEntity<ResponseObject> getActivityById(Integer activityId);
    
    ResponseEntity<ResponseObject> getActivitiesByScheduleId(Integer scheduleId);
    
    ResponseEntity<ResponseObject> getActivitiesByClassId(Integer classId);
    
    ResponseEntity<ResponseObject> getAllActivities();
    
    ResponseEntity<ResponseObject> deleteActivity(Integer activityId);
    
    ResponseEntity<ResponseObject> assignActivityToClass(AssignActivityToClassRequest request);
    
    ResponseEntity<ResponseObject> getActivitiesByClassAndDay(Integer classId, String dayOfWeek);
}